### ISSUE TITLE
Pull-requests: when, where, what, how?

### ISSUE THREAD

**Issue by dmf95** [shannonhlo/pokemon_classification#8]
Let's talk about the process of creating a pull-request to ensure we are all on the same. Here are some starter questions to discuss:

* when... should we make a pull-request?
* where... should we make the pull-request?
* what... should a pull-request have as it relates to comments, code, naming convention, etc. ?
* how... should we notify others of a pull-request?
* how... often should we be creating / reviewing pull-requests?
* how... many reviewers should we make as the minimum # of reviewers for each branch (e.g. rules for `develop` and `main` branches)

**Comment by gavh3** [comment]
My 2 cents:
- **When** the minimum scope of the linked issue has been addressed, then start a PR.
- **Where** `develop` and not `main`, until a major release milestone is met.
- **What** should be included in the PR? At minimum, a TLDR explaining what was **added**. Anything **changed** should be given brief explanation on *why* it was changed.
- **How** should we notify others of a PR? Messenger or (recommended) tag us as reviewers
- **How** often should we be creating / reviewing PRs? No hard guidelines. Let's try to be courteous and make an effort to review a PR if it has been open for a week with no activity.
- I propose that *all* of us are tagged for all PRs (to encourage learning what changed) and that we try to get at minimum 50% of reviewers approve before merging. For `main` however, the approval should be unanimous.

**Comment by shannonhlo** [comment]
Agreed with the points above. My thoughts on **how should we notify others of a PR?** is to tag us as reviewers. I imagine this to be a problem only if folks don't use their email or have email notifications turned off for PRs.
